tests/page_tests.py

```diff
@@ -236,14 +243,17 @@ def get_url_basename(url):
             else:
                 return url
 
-        def check_link(lnk, sess, filenames, folders):
-
-            if not check_skip_conditions(lnk, filenames, folders):
-                failure = try_to_connect(lnk, sess)
+        def check_link(link, session, filenames, folders):
```

Comment:
Could you please add a pydoc for this method and comments for conditional statements inside it, so the algorithm is more clear?